executor/runtime/docker/docker.go

```diff
@@ -477,7 +477,12 @@ func (r *DockerRuntime) mainContainerDockerConfig(c runtimeTypes.Container, bind
 		Soft: ((c.Resources().Disk * MiB) + 1*GiB),
 		Hard: ((c.Resources().Disk * MiB) + 1*GiB),
 	}
-	hostCfg.Ulimits = []*units.Ulimit{coreLimit}
+	fdLimit := &units.Ulimit{
+		Name: "nofile",
+		Soft: 100000,
```

Comment:
This seems low?

I swear I've changed this in the past, but I guess it didn't work because it must have been in some other codebase.

idk, of course the most important part is that there is *some* bound.
How high can we get away with, can we do 1mil ?

Reply:
The docker default is ~1M, so that's where things are currently at. In terms of what we can get away with, I don't know. The bug that triggered this PR was tasks leaking 400-700k fds, and nothing fell over. However, that is so far above what we usually use that the graphs looked scary. Maybe that's fine, though.

Reply:
100k is pretty high, it's about what some high-fd services would recommend for production:
- [kafka](https://docs.confluent.io/platform/current/kafka/deployment.html): "You should increase your file descriptor counttoat least 100,000."
- [cassandra](https://docs.datastax.com/en/cassandra-oss/3.x/cassandra/install/installRecommendSettings.html): "nofile 100000"

Is that 1M docker limit across all containers? If so then 100k/container would probably make sense to reduce the likelihood of multiple high-fd containers conflicting

Reply:
> Is that 1M docker limit across all containers? If so then 100k/container would probably make sense to reduce the likelihood of multiple high-fd containers conflicting

No, it is per container.